pkg/operator/deploymentcontroller/deployment_controller.go

```diff
@@ -303,6 +303,15 @@ func (c *DeploymentController) syncManaged(ctx context.Context, opSpec *opv1.Ope
 		status = status.WithConditions(progressingCondition)
 	}
 
+	// Degrade when operator is progressing too long.
+	progressingTooLong, err := v1helpers.IsUpdatingTooLong(opStatus, c.instanceName+opv1.OperatorStatusTypeProgressing)
```

Comment:
In addition to the above, Progressing condition is added only `if slices.Contains(c.conditions, opv1.OperatorStatusTypeProgressing` passes. It would be safer to add it into the `if` branch above.

Reply:
If we don't check the condition presence elsewhere, as I initially intended, then this is indeed better - moved.